### PR TITLE
Implement auth code flow for openshift login

### LINKED ISSUE #4890: Move OpenShift authentication to use the "authorization code" flow
The OpenShift authentication is currently using OAuth's _implicit flow_. For improved security, OpenShift auth should use the OAuth _authorization code_ flow.

### PR DESCRIPTION
**Pull request by nrfox**
### Describe the change

Moves the openshift auth strategy to authorization code flow. Adds autodiscovery of openshift auth options.

### Steps to test the PR

Regression testing with openshift auth strategy. Ensure login works properly.

### Automation testing

Covered by login cypress test. Added an additional test for showing login errors with openshift auth strategy.

### Issue reference

Fixes #4890

### CODE REVIEW

**Comment by nrfox** [comment]
Operator PR: https://github.com/kiali/kiali-operator/pull/760

File: business/authentication/openshift_auth_controller.go [hunk 1]
```diff
@@ -172,14 +262,15 @@ func (o openshiftAuthController) TerminateSession(r *http.Request, w http.Respon
...
-	if len(sPayload.Token) == 0 {
+	if len(sPayload.AccessToken) == 0 {
 		return TerminateSessionError{
 			Message:    "Cannot logout: the Openshift token is absent from the session",
 			HttpStatus: http.StatusInternalServerError,
 		}
 	}
 
-	err = o.openshiftOAuth.Logout(sPayload.Token)
+	// TODO: Support multi-cluster. A single logout should termiante all sessions.
```
Comment (jmazzitelli) [inline business/authentication/openshift_auth_controller.go:272]:
```suggestion
	// TODO: Support multi-cluster. A single logout should terminate all sessions.
```

**Comment by jmazzitelli** [comment]
I tested this and it works.

I checked out this PR, the [operator PR](https://github.com/kiali/kiali-operator/pull/760), and the [helm chart PR](https://github.com/kiali/helm-charts/pull/255). I started CRC, installed Istio, then I ran this (notice this also runs the unit tests, which also pass):

`make -e HELM_CHARTS_REPO_PULL=false clean clean-ui build build-ui test cluster-push operator-create kiali-create`

I was then able to log into Kiali using my OpenShift login credentials.

**Review by jmazzitelli — APPROVED** [review]
tested successfully

**Pull request merged**